modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpPipeliningHandler.java

```diff
@@ -227,8 +278,28 @@ private boolean doFlush(ChannelHandlerContext ctx) {
         }
         boolean needsFlush = true;
         while (channel.isWritable()) {
-            final WriteOperation currentWrite = queuedWrites.poll();
+            WriteOperation currentWrite = queuedWrites.poll();
             if (currentWrite == null) {
+                drainQueuedNowWritable(ctx);
+                if (channel.isWritable() == false) {
+                    // try flushing to make channel writable again, loop will only continue if channel becomes writable again
+                    ctx.flush();
+                    needsFlush = false;
+                }
+                if (channel.isWritable() == false) {
+                    break;
+                }
+                currentWrite = queuedWrites.poll();
+            }
+            if (currentWrite == null) {
+                // no write from a non-chunked message was found queued to be written, check if a chunked message might have become
```

Comment:
I think this comment is now misleading, since we also queue chunked writes?

Reply:
++ reworded